AnnService/inc/Core/BKT/Index.h

```diff
@@ -122,8 +122,8 @@ namespace SPTAG
                 return 1.0f - xy / (sqrt(xx) * sqrt(yy));
             }
             inline float ComputeDistance(const void* pX, const void* pY) const { return m_fComputeDistance((const T*)pX, (const T*)pY, m_pSamples.C()); }
-            inline const void* GetSample(const SizeType idx) const { return (void*)m_pSamples[idx]; }
-            inline bool ContainSample(const SizeType idx) const { return idx < m_pSamples.R() && !m_deletedID.Contains(idx); }
+            inline const void* GetSample(const SizeType idx) const { return (idx >= 0 && idx < m_pSamples.R()) ? (void*)m_pSamples[idx] : nullptr; }
```

Comment:
Philip, it is better not to do the check in the GetSample since it will be called many times in the search. Add the check in the ContainSample is ok. So if the wrapper want to get sample, check the contain sample first.